recipes/wineditline/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+import functools
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class WineditlineConan(ConanFile):
+    name = "wineditline"
+    description = (
+        "A BSD-licensed EditLine API implementation for the native "
+        "Windows Console"
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://mingweditline.sourceforge.net/"
+    topics = ("readline", "editline", "windows")
+    license = "BSD-3-Clause"
+    generators = ("cmake",)
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+    }
+    default_options = {
+        "shared": False,
+    }
+
+    exports_sources = ("CMakeLists.txt",)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Windows":
+            message = "wineditline is supported only on Windows."
+            raise ConanInvalidConfiguration(message)
+
+    def source(self):
+        root = self._source_subfolder
+        get_args = self.conan_data["sources"][self.version]
+        tools.get(**get_args, destination=root, strip_root=True)
+        for dir in ["bin32", "bin64", "include", "lib32", "lib64"]:
+            tools.rmdir(os.path.join(root, dir))
+        os.remove(os.path.join(root, "CMakeLists.txt"))
+        os.remove(os.path.join(root, "src", "CMakeLists.txt"))
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

Keep source untouched, remove what you need on build folder.